fdbserver/DDShardTracker.actor.cpp

```diff
@@ -368,23 +376,15 @@ ACTOR Future<Void> readHotDetector(DataDistributionTracker* self) {
 	try {
 		loop {
 			state KeyRange keys = waitNext(self->readHotShard.getFuture());
-			state Transaction tr(self->cx);
-			loop {
-				try {
-					Standalone<VectorRef<ReadHotRangeWithMetrics>> readHotRanges =
-					    wait(self->cx->getReadHotRanges(keys));
-					for (const auto& keyRange : readHotRanges) {
-						TraceEvent("ReadHotRangeLog")
-						    .detail("ReadDensity", keyRange.density)
-						    .detail("ReadBandwidth", keyRange.readBandwidth)
-						    .detail("ReadDensityThreshold", SERVER_KNOBS->SHARD_MAX_READ_DENSITY_RATIO)
-						    .detail("KeyRangeBegin", keyRange.keys.begin)
-						    .detail("KeyRangeEnd", keyRange.keys.end);
-					}
-					break;
-				} catch (Error& e) {
-					wait(tr.onError(e));
-				}
+			Standalone<VectorRef<ReadHotRangeWithMetrics>> readHotRanges = wait(self->db->getReadHotRanges(keys));
```

Comment:
Does this throw any retryable errors? With this change, such an error would cause the function to throw rather than retry. We should check to what errors are possible.

Reply:
I think the implementation of `getReadHotRanges` and `waitStorageMetrics` already have `try catch` block to handle the possible errors. 
And looking at the `tr.onError()` code, the retriable errors defined in it won't be thrown in a code not using a transaction IMO.

Reply:
I don't see errors clearly categorized as such in onError, but could we add an assertion statement to validate your assessment and to defend against future violations of that?

Reply:
If there are no such errors being thrown, then there shouldn't be a behavior change happening here and that's probably fine. I'm not sure if it makes sense to add the assert or not; it's hard to say for certain what the future expectation will be if somebody adds new errors in this path.

Reply:
The assertion is not that necessary I think, because if a retriable error is thrown, that would cause DD throw that error and then cause a simulation failure. 

BTW, I think it would be useful to have some method defined in the error class of which error is retriable under certain circumstances. Now we just judge the fact from reading the code.

Reply:
In my experience, the assertion serves as documentation of conditions believed to be true and decision such as this where a conscious choice is made not to handle the family of errors. In a few weeks, months or years, if and when a retry-able error does come into the picture here:
(1) the person who sees it is going to wonder why the test failed later than it could have
(2) the person is going to wonder why there is no loop there - whether that was a choice, or whether it was sloppy coding, or whether there is some kind of design consideration that makes it so this transaction should not be retried. At that point, we're banking on someones memory of this PR being good enough.